### PR TITLE
Created new notification component

### PR DESCRIPTION
**Pull request by joannaquu**
### Platforms Impacted
- [x ] iOS
- [ ] macOS
- [ ] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

Created the FURN notification component and updated the notification test to use it

### Verification

Screen Shot 2022-06-14 at 11 24 37 AM

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [x ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,9107 @@
+{
+  "name": "@fluentui-react-native/repo",
```
Comment (Saadnajmi) [inline package-lock.json:2]: Delete this file please. Did component-generator create it?

File: package.json [hunk 2]
```diff
@@ -82,5 +82,9 @@
     "react-native-windows": "^0.64.3",
     "react-native-macos": "^0.64.6",
     "simple-plist": "2.0.0-rc.0"
+  },
+  "dependencies": {
+    "gulp": "^4.0.2",
+    "gulp-rename": "^2.0.0"
```
Comment (Saadnajmi) [inline package.json:88]: why this change?

File: packages/components/Notification/jest.config.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+const { configureReactNativeJest } = require('@fluentui-react-native/scripts');
+// module.exports = configureReactNativeJest('android');
```
Comment (Saadnajmi) [inline packages/components/Notification/jest.config.js:2]: Tempte generated a commented out line?

File: packages/components/Notification/src/useNotification.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+export const useNotification = (props: NotificationProps): NotificationProps => {
+  const { variant, startText, endText, ...rest } = props;
+
+  return {
+    variant,
+    startText,
+    endText,
+    ...rest,
+  };
+};
```
Comment (Saadnajmi) [inline packages/components/Notification/src/useNotification.ts:12]: Lol, a very not useful hook at the moment :D. I guess we might as well keep it because I think it will change.

File: yarn.lock [hunk 5]
```diff
@@ -1407,6 +1407,24 @@
   resolved "https://registry.yarnpkg.com/@fluentui-react-native/design-tokens-windows/-/design-tokens-windows-0.13.0.tgz#1d8610634920bd5decaca669cae8183c0713d2ef"
   integrity sha512-o9ir+tpie+Fm9zO+nCbJLBfzVjUhugRWRsPS8n6wBaujC1qyeUYjhoGg6ejXzKc0nVCkIsFmRRRgC0brfrPIzw==
 
+"@fluentui-react-native/framework@0.7.17":
```
Comment (Saadnajmi) [inline yarn.lock:1410]: I think the dependencies in your Notication's package.json might be out of date. Can you make sure it's depending on the latest version of all the @fluentui-* packages? Their versions are just elsewhere in our repo, also in package.json files

File: package-lock.json (hunk 1 above)
Comment (joannaquu) [reply package-lock.json:2]: Yep

File: package.json (hunk 2 above)
Comment (joannaquu) [reply package.json:88]: This was from one of the commands run to use the component generator

Comment (Saadnajmi) [reply package.json:88]: I would also delete these from this change

File: package-lock.json (hunk 1 above)
Comment (Saadnajmi) [reply package-lock.json]: @ValentinaKozlova FYI. It seems like component-generator is using NPM instead of yarn, and therefore generating a package-lock.json? It's also adding dependencies to the root package.json. Otherwise, thanks for the awesome FHL project!

File: packages/components/Notification/jest.config.js (hunk 3 above)
Comment (Saadnajmi) [reply packages/components/Notification/jest.config.js]: Also @ValentinaKozlova , was this being commented out on purpose?

**Review by Saadnajmi — APPROVED** [review]

File: packages/components/Notification/jest.config.js (hunk 3 above)
Comment (Saadnajmi) [reply packages/components/Notification/jest.config.js]: JK, this is necessary. Jest tests fail without it :(

**Review by Saadnajmi — APPROVED** [review]

File: packages/components/Notification/src/NotificationTokens.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import { NotificationTokens } from './Notification.types';
+
+export const defaultNotificationTokens: TokenSettings<NotificationTokens, Theme> = (t: Theme) =>
+  ({
+    backgroundColor: 'skyblue',
+    color: t.colors.bodyText,
+    borderRadius: 12,
+    padding: 16,
+    neutral: {
+      backgroundColor: 'lightgrey',
```
Comment (rurikoaraki) [inline packages/components/Notification/src/NotificationTokens.ts:12]: Out of curiosity is there a plan to use design token driven values in the future?

I suppose this is meant for iOS so the issue is that there aren't any tokens to use yet?

File: packages/components/Notification/src/useNotification.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import { NotificationProps } from './Notification.types';
+
+export const useNotification = (props: NotificationProps): NotificationProps => {
+  const { variant, endText, ...rest } = props;
+
+  return {
```
Comment (rurikoaraki) [inline packages/components/Notification/src/useNotification.ts:6]: It doesn't seem like this function is needed, since it seems like the object is just being recreated but none of the properties are changing... Is there a plan to add more logic in here in a future change?

Comment (joannaquu) [reply packages/components/Notification/src/useNotification.ts:6]: This was created by the component generator and will most likely eventually contain more logic but it can be removed for this initial pr.

File: packages/components/Notification/src/NotificationTokens.ts (hunk 6 above)
Comment (Saadnajmi) [reply packages/components/Notification/src/NotificationTokens.ts:12]: The plan is to follow the FluentUI Apple control, which is tokenized, refers to some global/alias tokens, but isn't hooked up to the token-pipeline yet. I'm not familiar with the timeline to hook up FluentUI Apple to the token-pipeline, or stand up the iOS design tokens package. Because standing up the iOS design tokens package is out of scope for this project, we'll probably have to copy / hardcode most of the default tokens. I think there are some cases where we can refer to design tokens right now, since it looks like the FluentUI Apple control does. 

If you're curious, the WIP FluentUI Apple control's token definitions are here: https://github.com/microsoft/fluentui-apple/blob/fluent2-tokens/ios/FluentUI/Vnext/Notification/NotificationTokens.swift
You can see line 65 refers to globalTokens.brandColor.tint40, and line 156-160 refers to aliasTokens.shadow.shadow16

Comment (rurikoaraki) [reply packages/components/Notification/src/NotificationTokens.ts:12]: Sounds good, thanks!

File: packages/components/Notification/src/useNotification.ts (hunk 7 above)
Comment (rurikoaraki) [reply packages/components/Notification/src/useNotification.ts:6]: I think it's fine to check in as is if there's future plans for using the function.

**Review by rurikoaraki — APPROVED** [review]

File: packages/components/Notification/src/NotificationTokens.ts (hunk 6 above)
Comment (Saadnajmi) [reply packages/components/Notification/src/NotificationTokens.ts:12]: Ha, we both commented at the same time :D
